src/component/left-pane.js

```diff
@@ -27,32 +36,64 @@ class LeftPane extends Component {
 			userList = [];
 		});
 	}
+
+	handleSearchChange(e) {
+		const searchString = e.target.value;
+		this.setState({ searchString });
+	}
+
+	clearSearch() {
+		if (this.state.searchString !== '') this.setState({ searchString: '' });
+	}
+
 	render() {
+		const { userList, searchString } = this.state;
+		const { currentSelected, userSelected } = this.props;
+
+		const isSearchedUser = user =>
+			searchString === '' ||
+			user.toLowerCase().indexOf(searchString.toLowerCase()) !== -1;
+
 		return (
 			<div className={this.props.className}>
 				<div className="user-list-container">
 					<div className="user-search-wrapper">
 						<div className="user-search">
-							<input placeholder="Search User" id="user-search-input" />
-							<button className="user-search-button">
-								<Icon>search</Icon>
+							<input
+								placeholder="Search User"
+								id="user-search-input"
+								value={searchString}
+								onChange={this.handleSearchChange}
+							/>
+							<button className="user-search-button" onClick={this.clearSearch}>
+								<Icon>
+									{ searchString === ''
+										? 'search'
+										: 'close' }
+								</Icon>
 							</button>
 						</div>
 					</div>
-            
+
 					<List className="user-list">
-						{this.state.userList.map((user, index) => {
-							const className = user === this.props.currentSelected ? 'active' : null;
-							return (
-								<div  key={index} onClick={() => (this.props.userSelected(user))}>
-									<ListItem dense button className={className}>
-										<Avatar><Icon>person</Icon></Avatar>
-										<ListItemText  primary={user} />
-									</ListItem>
-									<Divider />
-								</div>
-							);
-						})}
+						{userList.reduce((acc, user) => {
+							const className = user === currentSelected ? 'active' : null;
+
+							if (isSearchedUser(user)) {
+								acc.push(
+									<div key={user} onClick={() => userSelected(user)}>
```

Comment:
```suggestion
									<div key={user} onClick={() => userSelected(user)}>
```

Because of `user` is unique, I've used it as key.